packages/core/lib/commands/debug.js

```diff
@@ -64,38 +64,35 @@ const command = {
       }
     ]
   },
-  run: function(options, done) {
+  run: async function (options) {
+    const { promisify } = require("util");
     const debugModule = require("debug");
     const debug = debugModule("lib:commands:debug");
 
-    const { Environment } = require("@truffle/environment");
+    const {Environment} = require("@truffle/environment");
     const Config = require("@truffle/config");
 
-    const { CLIDebugger } = require("../debug");
+    const {CLIDebugger} = require("../debug");
 
-    Promise.resolve()
-      .then(async () => {
-        const config = Config.detect(options);
-        await Environment.detect(config);
+    const config = Config.detect(options);
+    await Environment.detect(config);
 
-        const txHash = config._[0]; //may be undefined
-        if (config.fetchExternal && txHash === undefined) {
-          throw new Error(
-            "Fetch-external mode requires a specific transaction to debug"
-          );
-        }
-        if (config.compileTests) {
-          config.compileAll = true;
-        }
-        if (config.compileAll && config.compileNone) {
-          throw new Error(
-            "Incompatible options passed regarding what to compile"
-          );
-        }
-        return await new CLIDebugger(config, { txHash }).run();
-      })
-      .then(interpreter => interpreter.start(done))
-      .catch(done);
+    const txHash = config._[0]; //may be undefined
+    if (config.fetchExternal && txHash === undefined) {
+      throw new Error(
+        "Fetch-external mode requires a specific transaction to debug"
+      );
+    }
+    if (config.compileTests) {
+      config.compileAll = true;
+    }
+    if (config.compileAll && config.compileNone) {
+      throw new Error(
+        "Incompatible options passed regarding what to compile"
+      );
+    }
+    const interpreter = await new CLIDebugger(config, {txHash}).run();
+    return promisify(interpreter.start)();
```

Comment:
```suggestion
    return await promisify(interpreter.start)();
```

Reply:
Also does this need `interpreter.start.bind.interpreter`?

Reply:
And here's a suggestion with both changes
```suggestion
    return await promisify(interpreter.start.bind(interpreter))();
```